core/sr-arithmetic/src/traits.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
```

Comment:
This file should not exists at all. We have two options here: 

- extract sr-arithmatic in a way that it no longer depends on sr-primitives, so that SR-primitives can depend on it. This is afaik for sure doable and needs more effort and scaffolding I suppose 
- just put the arithmetic stuff in the sr-primitves but in their own sub-module for better organisation. 

I was actually glad that you did the former approach, and it is much better if it works out, but for sure not like this.